drivers/ethernet/eth_stm32_hal.c

```diff
@@ -394,37 +383,19 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 	buf_header->tx_buff.len = remaining_read;
 	buf_header->tx_buff.next = NULL;
 
-#else /* CONFIG_ETH_STM32_HAL_API_V2 */
-#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X)
-	dma_buffer = dma_tx_buffer[cur_tx_desc_idx];
 #else
 	dma_buffer = (uint8_t *)(dma_tx_desc->Buffer1Addr);
-#endif /* CONFIG_SOC_SERIES_STM32H7X || CONFIG_SOC_SERIES_STM32H5X */
 
 	if (net_pkt_read(pkt, dma_buffer, total_len)) {
 		res = -ENOBUFS;
 		goto error;
 	}
-
-#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X)
-	ETH_BufferTypeDef tx_buffer_def;
-
-	tx_buffer_def.buffer = dma_buffer;
-	tx_buffer_def.len = total_len;
-	tx_buffer_def.next = NULL;
-#endif /* CONFIG_SOC_SERIES_STM32H7X || CONFIG_SOC_SERIES_STM32H5X */
 #endif /* CONFIG_ETH_STM32_HAL_API_V2 */
 
-#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X) || \
-	defined(CONFIG_ETH_STM32_HAL_API_V2)
-
-	tx_config.Length = total_len;
 #if defined(CONFIG_ETH_STM32_HAL_API_V2)
+	tx_config.Length = total_len;
 	tx_config.pData = &ctx;
 	tx_config.TxBuffer = &dma_tx_buffer_header[ctx.first_tx_buffer_index].tx_buff;
-#else
-	tx_config.TxBuffer = &tx_buffer_def;
-#endif /* CONFIG_ETH_STM32_HAL_API_V2 */
```

Comment:
> Are you sure these lines should be removed? In non API V2 case, they will be missing and this makes the remaining part of the function API V2 specific, which doesn't look right. Or it should be added line 489.

Yes, it should be removed because it is included in the:

```c
#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X) || \
	defined(CONFIG_ETH_STM32_HAL_API_V2)
``` 
so the else is not handling the case of all non-API V2, but only the case of:

```c
#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X)
``` 
Now the V1 API support has been removed for STM32H7 and STM32H5, so there is no need for this.

Reply:
> Note also that the code under line 452 `#ifndef CONFIG_ETH_STM32_HAL_API_V2` should now be removed.

Yes, I will remove it.